futurex_openedx_extensions/dashboard/serializers.py

```diff
@@ -921,3 +922,22 @@ class AggregatedCountsSerializer(ReadOnlySerializer):
     all_tenants = AggregatedCountsAllTenantsSerializer()
     by_tenant = AggregatedCountsOneTenantSerializer(many=True)
     limited_access = serializers.BooleanField()
+
+
+class FileUploadSerializer(ReadOnlySerializer):
+    """
+    Serializer for handling the file upload request. It validates and serializes the input data.
+    """
+    file = serializers.FileField()
```

Comment:
are we sure that having `FileField` here does not conflict with the later `{'url': upload_file(storage_path, file), 'uuid': short_uuid},` ? will it upload the file twice, one to the root and another to the correct directory? I never used it this way, can you please double check?

You can also use `FileField(upload_to=method_name_to_return_the_correct_path)` and let django storages do the upload for you. The current `upload_file` function sets the file as private, which is against our interest with the theme files

Reply:
Working fine locally. I know what you are referring to.....FileField without upload-to will save file on default Media Root, that is right but it does it on instance.save() before that File type was InMemoryFile and in our case we are not calling instance.save and handling file manually so that's why it is working fine.

About Private files, I already handled that in upload_file function, It will only set files private if upload_file is receiving is_private param True.

Reply:
yes you're right